packages/panzer/disc-fe/src/evaluators/Panzer_ScatterResidual_Tpetra_impl.hpp

```diff
@@ -483,6 +512,36 @@ evaluateFields(typename TRAITS::EvalData workset)
 
 }
 
+// **********************************************************************
+template<typename TRAITS,typename LO,typename GO,typename NodeT>
+void panzer::ScatterResidual_Tpetra<panzer::Traits::Tangent, TRAITS,LO,GO,NodeT>::
+evaluateFields(typename TRAITS::EvalData workset)
+{
+  typedef TpetraLinearObjContainer<double,LO,GO,NodeT> LOC;
+
+  // for convenience pull out some objects from workset
+  std::string blockId = this->wda(workset).block_id;
+
+  Teuchos::RCP<typename LOC::VectorType> r = tpetraContainer_->get_f();
+
+  globalIndexer_->getElementLIDs(this->wda(workset).getLocalCellIDs(),scratch_lids_);
+
+  ScatterResidual_Tangent_Functor<ScalarT,LO,GO,NodeT> functor;
+  functor.r_data = r->getLocalViewDevice(Tpetra::Access::ReadWrite);
+  functor.lids = scratch_lids_;
+
+  // for each field, do a parallel for loop
+  for(std::size_t fieldIndex = 0; fieldIndex < scatterFields_.size(); fieldIndex++) {
+    functor.offsets = scratch_offsets_[fieldIndex];
+    functor.field = scatterFields_[fieldIndex];
+    functor.dfdp_fields = dfdpFieldsVoV_.getViewDevice();
+    // TODO BWR is there a better way? Is this correct?
+    functor.num_params = PHX::getFadSize(scatterFields_[fieldIndex].get_static_view())-1;
```

Comment:
@rppawlo Is this a good way to get the number of derivatives?

Reply:
@reuterb This is usually what I use
```
const int fadSize = Kokkos::dimension_scalar(field_.get_view());
```